components/kots-config-check/database/entrypoint.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+# Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+# Licensed under the GNU Affero General Public License (AGPL).
+# See License-AGPL.txt in the project root for license information.
+
+
+set -euo pipefail
+
+DB_IN_CLUSTER_ENABLED="${1:-""}"
+DB_CLOUDSQL_ENABLED="${2:-""}"
+DB_USERNAME="${3:-""}"
+DB_PASSWORD="${4:-""}"
+DB_HOST="${5:-""}"
+DB_PORT="${6:-""}"
+CSP_INSTANCES="${7:-""}"
+CSP_CREDENTIALS="${8:-""}"
+
+connection="false"
+version=""
+
+DB_TYPE="incluster"
+if [ "${DB_IN_CLUSTER_ENABLED}" == "0" ]; then
+  if [ "${DB_CLOUDSQL_ENABLED}" == "1"  ]; then
+    DB_TYPE="cloudsqlproxy"
+  else
+    DB_TYPE="external"
+  fi
+fi
+
+case "${DB_TYPE}" in
+  cloudsqlproxy | external)
+    if [ "${DB_TYPE}" = "cloudsqlproxy" ]; then
+      echo "Connecting to CloudSQLProxy"
+
+      CREDENTIALS_FILE="/tmp/credentials.json"
+      echo "${CSP_CREDENTIALS}" | base64 -d > "${CREDENTIALS_FILE}"
+
+      # Config overrides
+      DB_HOST="0.0.0.0"
+      DB_PORT="8080"
+
+      # This is a long-running process
+      cloud_sql_proxy \
+        --instances="${CSP_INSTANCES}=tcp:${DB_PORT}" \
+        -credential_file="${CREDENTIALS_FILE}" &
+
+      # Give it a chance to connect
+      sleep 5
+    else
+      echo "Using external database"
+    fi
+
+    # Check the database version
+    version_query=$(mysql \
+      --connect-timeout=5 \
+      --database=gitpod \
+      --user="${DB_USERNAME}" \
+      --password="${DB_PASSWORD}" \
+      --host="${DB_HOST}" \
+      --port="${DB_PORT}" \
+      --execute="SELECT VERSION();" \
+      --silent \
```

Comment:
Random thought: Do we still get enough log data with the `silent` flag for the support bundle or should we remove this flag?

Reply:
Yes. The `--silent` flag basically removes all the connection information. At this point, we're only interested in the version so we can assign it to the `version_query` variable. It is therefore correct that we have the `--silent` flag.

Also, there's actually no way of debugging a preflight check so we are limited to just putting the data in a known format so we can perform regex queries on it.